pyscriptjs/src/pyodide.ts

```diff
@@ -1,4 +1,181 @@
+    console.log('scripts queue set');
+});
+
+let appConfig_: AppConfig = {
+    autoclose_loader: true,
+};
+
+appConfig.subscribe((value: AppConfig) => {
+    if (value) {
+        appConfig_ = value;
+    }
+    console.log('config set!');
+});
+
+export class PyodideRuntime extends Runtime {
+    src = 'https://cdn.jsdelivr.net/pyodide/v0.20.0/full/pyodide.js';
+    name = 'pyodide-default';
+    lang = 'python';
+    interpreter: PyodideInterface;
+
+    async loadInterpreter(): Promise<void> {
+        console.log('creating pyodide runtime');
+        // eslint-disable-next-line
+        // @ts-ignore
+        this.interpreter = await loadPyodide({
+            stdout: console.log,
+            stderr: console.log,
+            fullStdLib: false,
+        });
+
+        // now that we loaded, add additional convenience functions
+        console.log('loading micropip');
+        await this.loadPackage('micropip');
+
+        console.log('loading pyscript...');
+        const output = await this.runCodeAsync(pyscript);
+        if (output !== undefined) {
+            console.log(output);
+        }
+
+        console.log('done setting up environment');
+    }
+
+    runCode(code: string): any {
+        return this.interpreter.runPython(code);
+    }
+
+    async runCodeAsync(code: string): Promise<any> {
+        return await this.interpreter.runPythonAsync(code);
+    }
+
+    getGlobals(): any {
+        return this.interpreter.globals;
+    }
+
+    registerJsModule(name: string, module: object): void {
+        this.interpreter.registerJsModule(name, module);
+    }
+
+    async loadPackage(names: string | string[]): Promise<void> {
+        await this.interpreter.loadPackage(names);
+    }
+
+    async installPackage(package_name: string | string[]): Promise<void> {
+        if (package_name.length > 0){
+            const micropip = this.getGlobals().get('micropip');
+            await micropip.install(package_name);
+            micropip.destroy();
+        }
+    }
+
+    async loadFromFile(s: string): Promise<void> {
+        const filename = getLastPath(s);
+        await this.runCodeAsync(
+            `
+                from pyodide.http import pyfetch
+                from js import console
+
+                try:
+                    response = await pyfetch("${s}")
+                except Exception as err:
+                    console.warn("PyScript: Access to local files (using 'paths:' in py-env) is not available when directly opening a HTML file; you must use a webserver to serve the additional files. See https://github.com/pyscript/pyscript/issues/257#issuecomment-1119595062 on starting a simple webserver with Python.")
+                    raise(err)
+                content = await response.bytes()
+                with open("${filename}", "wb") as f:
+                    f.write(content)
+            `,
+        );
+    }
+
+    async initialize(): Promise<void> {
+        loader?.log('Loading runtime...');
```

Comment:
Seems like we could move some of these to Runtime.initialize but we can do this in a follow up PR. Just wanting to comment something that we can improve later

Reply:
Yeah, I also thought the same, but wanted to have a first pass on the PR first.

Reply:
moved `initializers`, `postInitializers`, `scriptsQueue`, etc. to `initialize` in super class 😄